app/Models/MenuLink.php

```diff
@@ -219,7 +219,12 @@ protected function getEntityRoute(): string
                 $route = $menuRoute;
             }
         }
-        return route($route, $this->getRouteParams());
+        $filter = '';
```

Comment:
Why not add this check in getRouteParams()? That's literally what the function does, add params to the route, and is a lot cleaner than adding the params as a string, and doesn't break when switching between `?` and `&` for the params to be added.